stage4.py

```diff
@@ -0,0 +1,48 @@
+from card import Card
+import random
+
+
+deck = list()
+cards_in_deck = 0
+is_card_identical = 0
+while cards_in_deck < 40:
+    pulled_card = Card()
+    for card in deck:
+        if card == pulled_card:
+            is_card_identical = 1
+            break
+    if is_card_identical == 1:
+        is_card_identical = 0
+        continue
+    else:
+        cards_in_deck += 1
+        deck.append(pulled_card)
+
+random.shuffle(deck)
+
+num_players = 0
+while int(num_players) not in range(2, 7):
+    num_players = input("Enter number of players (between 2 to 6): ")
+
+scores = list()
+for i in range(int(num_players)):
+    scores.append(0)
+
+player_turn = 0
+while deck:
+    pulled_card = deck.pop()
+    scores[player_turn] += pulled_card.num
+    player_turn += 1
+    if player_turn is num_players:
```

Comment:
x is y usually use to compare pointers, and it rarely used in python.  equal is better here (we would like to compare the value) 
` if player_turn == num_players:`

https://www.geeksforgeeks.org/difference-operator-python/